warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownUnindexedFieldsVisitor.java

```diff
@@ -0,0 +1,131 @@
+package datawave.query.jexl.visitors;
+
+import static datawave.query.jexl.nodes.QueryPropertyMarker.MarkerType.BOUNDED_RANGE;
+import static datawave.query.jexl.nodes.QueryPropertyMarker.MarkerType.EVALUATION_ONLY;
+
+import java.util.Set;
+
+import org.apache.commons.jexl3.parser.ASTAndNode;
+import org.apache.commons.jexl3.parser.ASTEQNode;
+import org.apache.commons.jexl3.parser.ASTERNode;
+import org.apache.commons.jexl3.parser.ASTReferenceExpression;
+import org.apache.commons.jexl3.parser.JexlNode;
+import org.apache.commons.jexl3.parser.JexlNodes;
+import org.apache.log4j.Logger;
+
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.LiteralRange;
+import datawave.query.jexl.nodes.QueryPropertyMarker;
+
+/**
+ * Visitor meant to 'push down' predicates for expressions that are not executable against the global OR field index.
+ */
+public class PushdownUnindexedFieldsVisitor extends RebuildingVisitor {
```

Comment:
Will you be adding any tests for this visitor?

Reply:
TBD

Reply:
Yes, will do